playwright/tests/validators.spec.ts

```diff
@@ -0,0 +1,35 @@
+import { test, expect } from '@playwright/test'
+import { privateKey, privateKeyAddress } from '../utils/test-inputs'
+import { fillPrivateKeyWithoutPassword } from '../utils/fillPrivateKey'
+import { warnSlowApi } from '../utils/warnSlowApi'
+import { mockApi } from '../utils/mockApi'
+import { expectNoErrorsInConsole } from '../utils/expectNoErrorsInConsole'
+
+test.beforeEach(async ({ page }) => {
+  await warnSlowApi(page)
+  await mockApi(page, 500000000000)
+  await expectNoErrorsInConsole(page)
+
+  await page.route('**/validator/list?*', route => {
+    route.continue()
+  })
+
+  await page.goto('/open-wallet/private-key')
+  await fillPrivateKeyWithoutPassword(page, {
+    privateKey: privateKey,
+    privateKeyAddress: privateKeyAddress,
+    persistenceCheckboxChecked: false,
+    persistenceCheckboxDisabled: false,
+  })
+  await expect(page.getByTestId('account-selector')).toBeVisible()
+  await page.getByRole('link', { name: 'Stake' }).click()
+  await expect(page.getByRole('heading', { name: 'Validators' })).toBeVisible()
+})
+
+test.describe('Validators', () => {
+  test('Content-Security-Policy should allow validator icons', async ({ page, baseURL }) => {
+    expect(baseURL).toBe('http://localhost:5000')
+    expect((await page.request.head('/')).headers()).toHaveProperty('content-security-policy')
+    await expect(page.getByText('Everstake')).toBeVisible()
```

Comment:
This only checks the text, but not the icon itself? Or am I missing something?
Only way to check would be to take the screenshot.

Reply:
This waits for validator list to be loaded and `await expectNoErrorsInConsole(page)` above continuously checks there have been no CSP violations so far.

But I'll check if there's an explicit way to check "is image loaded" too!

Reply:
Thank u